### DISCUSSION TITLE
Caddy?

### DISCUSSION THREAD

**Discussion by leedickey** [tteck/Proxmox#2853] [Ideas]
What would it take to get Caddy as an installer script?  I am not a fan of Traefik and NPM.   I do find the simplicity of Caddy preferable as a reverse proxy.

**Comment by tteck** [comment]
Done

**Reply by leedickey** [reply]
Fantastic! Thank you!

**Reply by millercentral** [reply]
Does the Caddy LXC include the cloudflare dns plugin by chance?

**Comment by di5cord20** [comment]
I'm new to Caddy and looking for some help.  I'd like to create a local reverse proxy (i think) for my sonarr container (created through tteck scipts as well).  I'm able to reach the sonarr url I added to my Caddyfile, but the cert isn't valid.  Hoping someone can help or point me in the right direction.

My Caddyfile:
```
sonarr.lan {
    reverse_proxy sonnar-lxc-local-ip:8989
}
```
In case it helps, I'm seeing the following when I `cat /var/mail/root`:
```
From caddy@caddy.localdomain  Sun May 12 20:49:04 2024
Return-Path: <caddy@caddy.localdomain>
X-Original-To: root
Delivered-To: root@caddy.localdomain
Received: by caddy.localdomain (Postfix, from userid 999)
        id 0B2C66056; Sun, 12 May 2024 20:49:04 -0400 (EDT)
To: root@caddy.localdomain
From: caddy@caddy.localdomain
Auto-Submitted: auto-generated
Subject: *** SECURITY information for caddy.local ***
Message-Id: <20240513004904.0B2C66056@caddy.localdomain>
Date: Sun, 12 May 2024 20:49:04 -0400 (EDT)

caddy.local : May 12 20:49:04 : caddy : user NOT in sudoers ; PWD=/ ; USER=root ; COMMAND=/usr/bin/tee /usr/local/share/ca-certificates/Caddy_Local_Authority_-_2024_ECC_Root_225282169852296556828811619719316968955.crt
```

**Reply by leedickey** [reply]
Must be a valid domain I believe. They will not issue certs for any domain that you do not own.

**Reply by di5cord20** [reply]
hmm, perhaps I misunderstood.  I thought one of the unique benefits of Caddy was that this was possible.  I do have a domain and have a working traefik instance running in Docker.  I wanted to give Caddy a try though for some time and with the new tteck script, thought i would give it a go.

**Comment by rpp134** [comment]
I apologize if this is obvious but I can't figure out what the user name or password is for the LXC. Again, I don't know how to create the file without using the shell. Am I totally doing this wrong?
